taichi/backends/vulkan/spirv_ir_builder.cpp

```diff
@@ -39,20 +36,13 @@ void IRBuilder::init_header(bool support_int8,
         .commit(&header_);
   }
 
-  this->support_int8_ = support_int8;
-  if (support_int8) {
-    ib_.begin(spv::OpCapability).add(spv::CapabilityInt8).commit(&header_);
-  }
-  this->support_int16_ = support_int16;
-  if (support_int16) {
+  if (vulkan_cap_.has_int16) {
```

Comment:
Maybe keep int8 types? GPUs like AMD's RDNA2 has a very good int8 performance.

Reply:
`Int8` should enable the `Kernel` capability in SPIR-V

Reply:
We can still have 8 bit storage and pointers to 8 bit elements through `VK_KHR_8bit_storage` and `SPV_KHR_8bit_storage`. I'm not sure whether we should mark it as support int8 tho since its sole purpose is storage not registers. I guess this would be fine.

Reply:
Ah also Kernel capability is not a requirement now. According to SPIR-V spec:
"Khronos SPIR-V Issue #285: Remove Kernel as a capability implicitly declared by Int8."

As long as the hardware support it we should enable it

Reply:
It makes sense. I'll fix it.

Reply:
Fixed